File: src/main/java/io/pivotal/pal/tracker/TimeEntryRepository.java
package io.pivotal.pal.tracker;

import java.util.List;

public interface TimeEntryRepository {
  TimeEntry create(TimeEntry timeEntry);
  TimeEntry update(long id, TimeEntry timeEntry);
  TimeEntry delete(long id);
  TimeEntry find(long id);
  List<TimeEntry> list();
}

File: src/main/kotlin/io/pivotal/pal/tracker/TimeEntryRepository.kt
package io.pivotal.pal.tracker

interface TimeEntryRepository {
    fun create(timeEntry:TimeEntry):TimeEntry
    fun find(id:Long): TimeEntry?
    fun list():List<TimeEntry>
    fun update(id:Long, timeEntry:TimeEntry):TimeEntry?
    fun delete(id:Long):TimeEntry?
}